charts/karpenter/templates/rbac.yaml

```diff
@@ -1,12 +1,14 @@
+{{- if .Values.serviceAccount.create }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: karpenter
-  namespace: karpenter
+  namespace: {{ .Release.Namespace }}
   {{- with .Values.serviceAccount.annotations }}
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+{{- end }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
```

Comment:
I think we should include the clusterrolebinding inside the create flag, since the referenced ServiceAccount doesn't exist. WDYT?

Reply:
I'm thinking that if the SA is configured via other mechanisms (TF, eksctl, etc), we should probably template the SA name to allow customization and keep the ClusterRoleBinding, similar to how it's done in cluster autoscaler ([ref](https://github.com/helm/charts/blob/fd80db1fed08f4e4b9555d739a10f0242fec34bf/stable/cluster-autoscaler/values.yaml#L121)).

Adding a commit for that change here and adjusting PR description